example/codespaces/newusersecretwithxcrypto/main.go

```diff
@@ -0,0 +1,169 @@
+//
+// To verify the new secret, navigate to GitHub User > Settings > left side options bar > Codespaces > Secrets.
+//
+// Usage:
+//
+//	export GITHUB_AUTH_TOKEN=<auth token from github that has secret create rights>
+//	export SECRET_VARIABLE=<secret value of the secret variable>
+//	go run main.go SECRET_VARIABLE
+//
+// Example:
+//
+//	export GITHUB_AUTH_TOKEN=0000000000000000
+//	export SECRET_VARIABLE="my-secret"
+//	go run main.go SECRET_VARIABLE
+package main
+
+import (
+	"context"
+	crypto_rand "crypto/rand"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/google/go-github/v53/github"
+	"golang.org/x/crypto/nacl/box"
+)
+
+var (
+	repo  = flag.String("repo", "", "The repo that the secret should be added to, ex. go-github")
+	owner = flag.String("owner", "", "The owner of there repo this should be added to, ex. google")
+)
+
+func main() {
+	flag.Parse()
+
+	token := os.Getenv("GITHUB_AUTH_TOKEN")
+	if token == "" {
+		log.Fatal("please provide a GitHub API token via env variable GITHUB_AUTH_TOKEN")
+	}
+
+	secretName, err := getSecretName()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	secretValue, err := getSecretValue(secretName)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	ctx := context.Background()
+	client := github.NewTokenClient(ctx, token)
+
+	if err := addUserSecret(ctx, client, secretName, secretValue, *owner, *repo); err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("Added secret %q to the authenticated user\n", secretName)
+}
+
+func getSecretName() (string, error) {
+	secretName := flag.Arg(0)
+	if secretName == "" {
+		return "", fmt.Errorf("missing argument secret name")
+	}
+	return secretName, nil
+}
+
+func getSecretValue(secretName string) (string, error) {
+	secretValue := os.Getenv(secretName)
+	if secretValue == "" {
+		return "", fmt.Errorf("secret value not found under env variable %q", secretName)
+	}
+	return secretValue, nil
+}
+
+// addUserSecret will add a secret to a GitHub user for use in GitHub Codespaces.
+//
+// Finally, the secretName and secretValue will determine the name of the secret added and it's corresponding value.
+//
+// The actual transmission of the secret value to GitHub using the api requires that the secret value is encrypted
+// using the public key of the target user. This encryption is done using x/crypto/nacl/box.
+//
+// First, the public key of the user is retrieved. The public key comes base64
+// encoded, so it must be decoded prior to use.
+//
+// Second, the decode key is converted into a fixed size byte array.
+//
+// Third, the secret value is converted into a slice of bytes.
+//
+// Fourth, the secret is encrypted with box.SealAnonymous using the user's decoded public key.
+//
+// Fifth, the encrypted secret is encoded as a base64 string to be used in a github.EncodedSecret type.
+//
+// Sixt, The other two properties of the github.EncodedSecret type are determined. The name of the secret to be added
```

Comment:
```suggestion
// Sixth, The other two properties of the github.EncodedSecret type are determined. The name of the secret to be added
```

Reply:
Thanks. Copy paste error from the actions secrets example.